source/partial-on-undefined-deep.d.ts

```diff
@@ -68,3 +70,8 @@ type PartialOnUndefinedDeepValue<T, Options extends PartialOnUndefinedDeepOption
 		: T extends Record<any, any> | undefined
 			? PartialOnUndefinedDeep<T, Options>
 			: unknown;
+
+/**
+Utility type to retrieve only literal keys from type.
+*/
+type LiteralKeyOf<T> = keyof {[K in keyof T as IsLiteral<K> extends true ? K : never]-?: never};
```

Comment:
Can you move this to internal.d.ts? It makes it easier for us to keep track of what should probably be exported publicly.